[9.0] knowledge

naousse opened the pull request:
The knowledge module has been deprecated by Odoo SA in version 9.0. This pull request is a port of this module to provide:
- a top-level menu item
- submenus with a settings panel to configure knowledge and document applications
- groups and access rights

naousse committed:
knowledge module added from Odoo 8.0 and ported to 9.0

pedrobaeza commented:
But why do you need this module? This module was only a wrapper to allow certain configuration, but per se doesn't add any model. I don't think it's needed.

naousse commented:
OK. 
The module document_page need it, and it is the module that add the menu for knowledge operation.

naousse commented:
So, do i remove it as requirement for document_page module?

pedrobaeza commented:
Yes, I think it's the way to go. See if in current v9 menu layout, document_page menu entries fit in any place, or create a new root in it. To assure compatibility with another document_\* modules, you can put as XML-ID for this menu "base.menu_document" or similar, and declare the same menu in all modules that need it.

naousse commented:
OK! Thanks, i will go that way

dreispt commented:
@pedrobaeza But why create another module adding a new menu root when this module already does that? IMO it's not a problem to keep this, and `document_page`can depend on it for the creation of the top level menu.

pedrobaeza commented:
No, you don't need another module, Daniel. You only need to declare the menu on subsequent modules. This is the way to implement it: https://github.com/OCA/knowledge/pull/76/files#diff-1b36a8a2d376367ab8f27aac507585a3R5

I also ask if current v9 menu layout  can host menu entries for document_page and the rest of the document_\* modules without creating a parent menu.

max3903 commented:
@pedrobaeza I tend to agree with @dreispt: we should declare the menuitem once in knowledge module instead of every submodules (document_page, document_*)

There is no menu_document in base.

max3903 commented:
@naousse Same remarks as described here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#todo

and add `Odoo Community Association (OCA)` in the author keyword

pedrobaeza commented:
OK, as you want. I was saying that for avoiding a dumb module. The technique I mention is used on Odoo for Sales, Purchases, and other top menus.

naousse committed:
Knowledge fixed as base module for knowledge management system
reamed.rst added
version number corrected
res_config.py is now  respecting the new odoo API

naousse committed:
Edit readme.rst and models/res_config.py to solve previous issues

naousse committed:
license key added in __openerp__.py

naousse committed:
flake8 issues fixed in __openerp__py files and knowledge/models/res_config.py

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,104 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
https://runbot.odoo-community.org/runbot/repo/118/9.0

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,104 @@
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
```
These comments can be removed.

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,104 @@
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
+{module_name}%0Aversion:%20
+{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Replace {project_repo}, {module_name} and {version}

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,104 @@
+Configuration
+=============
+
+No configuration required
```
Go to Knowledge / Configuration / Settings

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,104 @@
+
+This module is an official module but was not present in odoo 9.0 official repository
+so we took it from Odoo 8.0 repository and add it here.
+
```
@naousse This is a sufficient description for the changelog. The rest below can be found in the commits and should be removed. Maybe only mention that you made changes to comply with OCA Conventions.

max3903 commented on knowledge/views/res_config.xml:
```diff
@@ -0,0 +1,38 @@
+            <field name="view_mode">form</field>
+            <field name="target">inline</field>
+        </record>
+        <menuitem id="menu_knowledge_configuration" name="settings" parent="knowledge.menu_document_configuration"
```
name="Settings"

naousse committed:
Readme correction

naousse commented:
I have made the corrections

max3903 commented:
:+1:

naousse committed:
License Errore Fixed

dreispt commented on knowledge/README.rst:
```diff
@@ -0,0 +1,78 @@
+
+Known issues / Roadmap
+======================
+
```
* Add options for other OCA extension modules as they are made available.

dreispt commented:
Other than a couple of minor notes, it's :+1:

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,78 @@
+Knowledge Management System base
+================================
+
+This module allows you to write web pages for internal documentation.
```
This module is the base for any knowledge and document management application.

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,78 @@
+Configuration
+=============
+
+Go to Knowledge / Configuration / Settings.
```
`None`

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,78 @@
+Usage
+=====
+
+To use this module, you need to:
```
* go to Knowledge / Configuration / Settings

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,78 @@
+============
+
+Makes the Knowledge Application Configuration available from where you can install
+document and Wiki based Hidden.
```
`None`

naousse committed:
Correction des erreurs pylint-odoo

naousse committed:
Correction of Readme typo

naousse committed:
correction typo

max3903 commented:
@OCA/tools-maintainers can anyone help us fixing the issue with Travis here?

max3903 commented on knowledge/__init__.py:
```diff
@@ -0,0 +1,23 @@
+"""Import of model's module."""
```
@naousse Please remove this line, it makes Travis failed.

gurneyalex replied:
@max3903 are you sure of this?

max3903 replied:
@gurneyalex Not this one, but in `__openerp__.py`, it does.

max3903 commented on knowledge/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+"""Knowledge Management System module."""
```
@naousse Same here.

gurneyalex replied:
this one probably, linked to this change: 

https://github.com/OCA/maintainer-quality-tools/commit/d8f265b93039a7f0199dcea3a3b92998db62def6

max3903 commented on knowledge/README.rst:
```diff
@@ -0,0 +1,77 @@
+Configuration
+=============
+
+* Go to Knowledge / Configuration / Settings
```
`* None`

gurneyalex commented:
I'm surprised by the WARNINGs in runbot 
https://runbot.odoo-community.org/runbot/build/3119551

gurneyalex commented:
https://github.com/OCA/maintainer-quality-tools/issues/295

naousse committed:
flake8 error fixed

naousse committed:
readme error fixed

naousse committed:
additional config settings added

max3903 commented:
:+1:
